src/TezosProvider.ts

```diff
@@ -0,0 +1,509 @@
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  // Method to send a delegation
+  public tezosSendDelegation = async (
+    op: PartialTezosDelegationOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  // Method to send an undelegation
+  public tezosSendUndelegation = async (): Promise<TezosSendResponse> => {
+    const op: PartialTezosDelegationOperation = { kind: TezosOperationType.DELEGATION };
+    return await this.tezosSend(op);
+  };
+
+  // Method to originate a contract
+  public tezosSendOrigination = async (
+    op: PartialTezosOriginationOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  // Method to call a smart contract: destination is the contract address, entrypoint as defined in the contract
+  public tezosSendContractCall = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendStake = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({
+      ...op,
+      destination: this.address,
+      parameters: { entrypoint: "stake", value: { prim: "Unit" } },
+    });
+  };
+
+  public tezosSendUnstake = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({
+      ...op,
+      destination: this.address,
+      parameters: { entrypoint: "unstake", value: { prim: "Unit" } },
+    });
+  };
+
+  public tezosSendFinalizeUnstake = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({
+      ...op,
+      destination: this.address,
+      parameters: { entrypoint: "finalize_unstake", value: { prim: "Unit" } },
+    });
+  };
+
+  public tezosSendActivateAccount = async (
+    op: TezosActivateAccountOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({ ...op, pkh: this.address });
+  };
+
+  public tezosSendBallot = async (op: TezosBallotOperation): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendDalPublishCommitment = async (
+    op: PartialTezosDalPublishCommitmentOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendFailingNoop = async (
+    op: TezosFailingNoopOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendIncreasePaidStorage = async (
```

Comment:
there is no need to await unless you want to catch errors or something. so it should be
```typescript
public async sendIncreasePaidStorage(op: PartialTezosIncreasePaidStorageOperation) {
  return this.tezosSend(op);
}
```